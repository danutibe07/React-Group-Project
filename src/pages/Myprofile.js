import React from 'react';
import { useSelector } from 'react-redux';
import Table from 'react-bootstrap/Table';

const Myprofile = () => {
  const { missions } = useSelector((store) => store.missions);

  return (
    <div className="profile-content">
      <div className="missions-list">
        <h2>My Missions</h2>
        <Table striped bordered>
          <tbody>
            {missions.filter((mission) => mission.reserved).map(
              (reserved) => (<tr key={reserved.id}><td>{reserved.name}</td></tr>),
            )}
          </tbody>
        </Table>
      </div>
    </div>
  );
};

export default Myprofile;
