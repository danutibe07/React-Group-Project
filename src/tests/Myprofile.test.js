import Myprofile from '../pages/Myprofile';

describe('Testing the profile', () => {
  const component = (
    <Myprofile mission={{
      mission_id: '1',
      mission_name: 'Test ',
      description: 'Test  description',
      reserved: true,
    }}
    />
  );

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
