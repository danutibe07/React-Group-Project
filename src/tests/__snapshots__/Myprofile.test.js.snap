// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders My Profile correctly 1`] = `
<div
  className="profile-content"
>
  <div
    className="missions-list"
  >
    <h2>
      My Missions
    </h2>
    <table
      className="table table-striped table-bordered"
    >
      <tbody />
    </table>
  </div>
</div>
`;
